subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/tasks/XcTest.java

```diff
@@ -50,15 +52,15 @@ public XcTest(ObjectFactory objectFactory) {
 
     @InputDirectory
     public File getTestBundleDir() {
-        return ((DirectoryVar) getExtensions().getExtraProperties().get("testBundleDir")).getAsFile().get();
+        return ((RegularFileVar) getExtensions().getExtraProperties().get("testBundleDir")).getAsFile().get();
     }
 
     public void setTestBundleDir(File testBundleDir) {
-        ((DirectoryVar) getExtensions().getExtraProperties().get("testBundleDir")).set(testBundleDir);
+        ((RegularFileVar) getExtensions().getExtraProperties().get("testBundleDir")).set(testBundleDir);
     }
 
-    public void setTestBundleDir(Provider<? extends Directory> testBundleDir) {
-        ((DirectoryVar) getExtensions().getExtraProperties().get("testBundleDir")).set(testBundleDir);
+    public void setTestBundleDir(Provider<? extends RegularFile> testBundleDir) {
```

Comment:
Why did this change from a Directory to a RegularFile? it seems like it's still a directory.

Reply:
See the other comment on this.